pennylane/ops/op_math/condition.py

```diff
@@ -225,7 +226,9 @@ def __call_capture_enabled(self, *args, **kwargs):
         @wraps(self.true_fn)
         def new_wrapper(*args, **kwargs):
 
-            jaxpr_true = jax.make_jaxpr(functools.partial(self.true_fn, **kwargs))(*args)
+            flat_args, in_tree = jax.tree_util.tree_flatten(args)
+            flat_fn = FlatFn(functools.partial(self.true_fn, **kwargs), in_tree)
+            jaxpr_true = jax.make_jaxpr(flat_fn)(*flat_args)

```

Comment:
The elif and false functions are not being flattened because we just need the in and out trees to figure out how to flatten/unflatten the inputs/outputs, right? Since the jaxpr for the flattened/unflattened functions will be the same?

But then why is the jaxpr for the true_fn made with the flattened function?

Reply:
So it turns out `jax.make_jaxpr` does flattening of inputs for us, so we don't need to pass flat args, and we don't need `FlatFn` to accept flat args.  So that's now been updated.

We only need one `FlatFn` to be used to calculate the required `out_tree`.  We will assume all functions have the same `out_tree` for now.  We could probably start explicitly verifying it once it becomes a problem.

Reply:
> We will assume all functions have the same out_tree for now. We could probably start explicitly verifying it once it becomes a problem.

The abstract eval for cond has logic baked in to verify that the out_aval of all branches is the same. That doesn't perfectly translate to the outputs also having the same `PytreeDef`, but I'm wondering if it's worth just making that assumption if flattening/unflattening to get the `out_tree` ends up being expensive.

Reply:
We add can in more validation if the lack turns out to cause problems.